packages/core/src/execute/BaseRoutine.ts

```diff
@@ -38,19 +38,38 @@ export default abstract class BaseRoutine<Ctx extends Context<BaseContextArgs>>
 
   async execute(context: Ctx): Promise<any[]> {
     const value = await this.serializeTasks();
-    const { other, priority } = this.orderByWorkspacePriorityGraph();
-
-    await this.serializeRoutines(value, priority);
+    const batches = this.orderByWorkspacePriorityGraph();
+    const errors: Error[] = [];
 
     const concurrency = context.args.concurrency || this.tool.config.execute.concurrency;
-    const response = await this.poolRoutines(value, concurrency ? { concurrency } : {}, other);
+    const responses: AggregatedResponse[] = [];
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const batch of batches) {
+      // eslint-disable-next-line no-await-in-loop
+      const response = await this.poolRoutines(value, concurrency ? { concurrency } : {}, batch);
+
+      responses.push(response);
```

Comment:
I wonder if we can ditch `responses` entirely and just store the results?

Reply:
good call. updated